nbs/src/tbats.ipynb

```diff
@@ -1005,34 +1013,40 @@
     "        y = y[max_index+1:len(y)]\n",
     "\n",
     "    # Check if there are negative values\n",
-    "    if np.any(y < 0): \n",
-    "        use_boxcox = False \n",
-    "\n",
-    "    # Check if there is a trend\n",
-    "    adf = adfuller(y, regression = 'ct')\n",
-    "    if adf[1] <= 0.05: \n",
-    "        warnings.warn('The time series is trend-stationary, disabling trend components.')\n",
-    "        use_trend = False\n",
-    "        use_damped_trend = False\n",
-    "    else: \n",
-    "        use_trend = True\n",
-    "\n",
-    "    # Choose the number of harmonics \n",
-    "    k_vector = np.array([find_harmonics(y, m) for m in seasonal_periods])\n",
+    "    if use_boxcox in (True, None) and np.any(y <= 0):\n",
+    "        warnings.warn(\"Data contains zero or negative values, disabling Box-Cox transformation.\")\n",
+    "        use_boxcox = False\n",
+    "\n",
+    "    # Choose the number of harmonics\n",
+    "    ks = []\n",
+    "    z = y.copy()\n",
+    "    for period in seasonal_periods:\n",
+    "        k, z = find_harmonics(z, period)\n",
+    "        ks.append(k)\n",
+    "    k_vector = np.array(ks)\n",
     "\n",
     "    # Select combinations to try out \n",
     "    if use_boxcox is None: \n",
     "        B = [True, False]\n",
     "    else: \n",
     "        B = [use_boxcox]\n",
     "\n",
-    "    if use_trend: \n",
-    "        if use_damped_trend is None: \n",
+    "    if use_trend is None:\n",
```

Comment:
Some models were undefined. For example, when using `use_trend=True` and `use_damped_trend=None`, one of the resulting combinations was [True, None]. As a result, not all possible models were being considered.